src/rgw/rgw_rest_s3.cc

```diff
@@ -3574,18 +3574,25 @@ RGWOp *RGWHandler_REST_Service_S3::op_post()
 RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
 {
   // Non-website mode
-  int list_type = 1;
+  int list_type;
   s->info.args.get_int("list-type", &list_type, 1);
 
    // Non-website mode    // Non-website mode
   if (get_data) {   
     if (list_type == 1) {
-       return new RGWListBucket_ObjStore_S3;     
-    } else if(list_type == 2) {
+      return new RGWListBucket_ObjStore_S3;
+    }
+    else if (list_type == 2) {
       return new RGWListBucket_ObjStore_S3v2;
-    } } else {
-    return new RGWStatBucket_ObjStore_S3;    
-  }   }
+    }
+    else {
+      dout(10) << "Unsupported list-type %d" << list_type;
```

Comment:
you might need to use `s` for sending the logging messages. see the definition of `RGWHandler`. like
```c++
ldpp_dout(s, 5) << __func__ << ": unsupported list-type " << list_type << dendl;
```

where i made 4 changes

1. print `__func__`. so we can pinpoint the place where the log message is printed.
1. use debug level 5 instead of 10, please see other warning messages we print when seeing bad requests in the same file.
1. remove "%d". i don't understand how the "printf()" format string can interact with the C++ ostream `operator<<`.
1. add `dend`. please refer the definition of `dout()`and `ldpp_dout()` macros for the reason.

Reply:
Oops, I did it in a hurry yesterday. Thanks.